evml/keras/models.py

```diff
@@ -983,9 +1157,33 @@ def predict_monte_carlo(self, x, mc_forward_passes=10, batch_size=None):
         )  # shape (n_samples,)
         # Calculating mutual information across multiple MCD forward passes
         mutual_info = entropy - np.mean(
-            np.sum(-y_prob * np.log(y_prob + epsilon), axis=-1), axis=0
+            np.sum(-np.array(y_prob) * np.log(y_prob + epsilon), axis=-1), axis=0
         )  # shape (n_samples,)
         return pred_probs, aleatoric, epistemic, entropy, mutual_info
+    
+    def predict_ensemble(self, x, weight_locations, batch_size=None):
+        num_models = len(weight_locations)
+
+        # Initialize output_shape based on the first model's prediction
+        if num_models > 0:
+            first_model = self.model
+            first_model.load_weights(weight_locations[0])
+            first_prediction = self.predict(x, batch_size=batch_size)
+            output_shape = first_prediction.shape[1:]
+            predictions = np.empty((num_models,) + (x.shape[0],) + output_shape)
+            predictions[0] = first_prediction
+        else:
+            output_shape = ()  # Default shape if no models
+            predictions = np.empty((num_models,) + (x.shape[0],) + output_shape)
+
+        # Predict for the remaining models
+        for i, weight_location in enumerate(weight_locations[1:]):
+            model_instance = self.model
+            model_instance.load_weights(weight_location)
+            y_prob = model_instance.predict(x, batch_size=batch_size)
+            predictions[i + 1] = y_prob
+
+        return predictions
 
     def compute_uncertainties(self, y_pred, num_classes=4):
         return calc_prob_uncertainty(y_pred, num_classes=num_classes)
```

Comment:
In the other model classes, I think the `calc_prob_uncertainties()` is a method within the class, and here it is a function outside the class. It would be better if they were consistent.